PySC packaging for PyPi

**Description, motivation and use case**
The following error appears when running tests on GitHub in dependent projects. This is due to the packaging of PySC, which is not published on PyPI (neither as an independent package nor bundled inside the PyAML package).

```bash
Run python -m pytest tests
ImportError while loading conftest '/home/runner/work/tango-pyaml/tango-pyaml/tests/conftest.py'.
tests/conftest.py:7: in <module>
    from tango.pyaml.controlsystem import ConfigModel as CsCM, TangoControlSystem
tango/pyaml/controlsystem.py:5: in <module>
    from pyaml.control.controlsystem import ControlSystem
/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/site-packages/pyaml/control/controlsystem.py:30: in <module>
    from ..tuning_tools.orbit import Orbit
/opt/hostedtoolcache/Python/3.14.2/x64/lib/python3.14/site-packages/pyaml/tuning_tools/orbit.py:19: in <module>
    from ..external.pySC.pySC import ResponseMatrix as pySC_ResponseMatrix
E   ModuleNotFoundError: No module named 'pyaml.external.pySC'
Error: Process completed with exit code 4.
```

**Proposed solution**
Package PySC inside the PyAML package. The exact implementation details will be defined later.

--- a/.github/workflows/deploy-pypi.yaml
+++ b/.github/workflows/deploy-pypi.yaml
@@ -17,6 +17,8 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
+      with:
+        submodules: recursive
     - name: Set up Python
       uses: actions/setup-python@v5
       with:

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -19,6 +19,8 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
+      with:
+        submodules: recursive
     - name: Set up Python 3.12
       uses: actions/setup-python@v3
       with:
@@ -26,14 +28,12 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        git submodule update --init --recursive
         pip install numpy
         pip install scipy
         pip install pydantic
         pip install accelerator-toolbox
         pip install matplotlib
         pip install h5py
-        pip install pyaml
         pip install flake8 pytest
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
     - name: Lint with flake8

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -34,6 +34,7 @@ jobs:
         pip install accelerator-toolbox
         pip install matplotlib
         pip install h5py
+        pip install PyYaml
         pip install flake8 pytest
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
     - name: Lint with flake8

--- a/pyaml/__init__.py
+++ b/pyaml/__init__.py
@@ -8,7 +8,7 @@ pyAML
 __title__ = "pyAML"
 __description__ = "Python Accelerator Middle Layer"
 __url__ = "https://github.com/python-accelerator-middle-layer/pyaml"
-__version__ = "0.2.1"
+__version__ = "0.2.2"
 __author__ = "pyAML collaboration"
 __author_email__ = ""
 

